pkgs/test_api/lib/src/backend/group.dart

```diff
@@ -96,8 +96,9 @@ class Group implements GroupEntry {
         metadata: metadata,
         trace: trace,
         location: location,
-        setUpAll: setUpAll,
-        tearDownAll: tearDownAll);
+        // Always clone these because they are being re-parented.
+        setUpAll: setUpAll?.filter((_) => true),
+        tearDownAll: tearDownAll?.filter((_) => true));
```

Comment:
This makes me wonder if we should just have a `clone()` as well as `.filter()`... I guess the code above for `onPlatform` could also just pass `(_) => true` to be consistent with this, but since the `setUpAll`/`tearDownAll` functions always have the same `platform` in the metadata as the enclosing group, it should always be the same.

Reply:
lol I left the same comment, it does feel wrong to do this just to clone

Reply:
Are there any implications of adding a new `clone()` to the `Test` class? (might anyone outside of here be `implement`ing it for example?)

Reply:
@jakemac53 I added a `clone()` in aef5437bd2754ee3eb2aed9af588ef7424d14b59 and updated the existing test `filter()`s to call it that were only there to clone anyway.

It does mean there are two almost identical implementations for `Group.filter()` and `Group.clone()` though (I did consider making `clone()` internally just call `filter((_) => true)` but I don't know how you'd feel about that!).

Reply:
> Are there any implications of adding a new clone() to the Test class? (might anyone outside of here be implementing it for example?)

It isn't part of the public API. If flutter is doing this, we can update it when the package rolls in (although it's possible this will make the google3 roll complicated).